File: Server/TrajectoryPoint.cs
public class TrajectoryPoint
{
    public GeoPoint Position { get; set; }
    public double Heading { get; set; }
    public double Pitch { get; set; }

    public TrajectoryPoint(GeoPoint position, double heading, double pitch)
    {
        Position = position;
        Heading = heading;
        Pitch = pitch;
    }
}

File: Client/src/js/msgTypes/TrajectoryPoint.js
export class TrajectoryPoint {
  constructor(position, heading, pitch) {
    this.position = position; // should be a GeoPoint instance
    this.heading = heading;
    this.pitch = pitch;
  }

  toString() {
    return `[${this.position.toString()}, Heading=${this.heading}, Pitch=${this.pitch}]`;
  }
}